### ISSUE TITLE
stack_snapshot ignores the cabal flags defined in the snapshot

### ISSUE THREAD

**Issue by regnat** [tweag/rules_haskell#1152]
**Describe the bug**

Stack snapshots can define custom cabal flags for building certain packages. The `stack_snapshot` rule ignores them.

**To Reproduce**

From the rules_haskell repository

1. Add `cassava` to the `packages` argument of `stack_snapshot` in the `WORKSPACE`
2. Build it with `bazel build @hackage//:cassava`

The build will fail with

```
Setup.hs: Encountered missing dependencies:
bytestring >=0.9.2 && <0.10.4, text-short ==0.1.*
```

Now add `flags = { "cassava": ["-bytestring--LT-0_10_4"], },` to the arguments of `stack_snapshot` and build it again. The build will succeed

**Expected behavior**

Cassava should be built with the correct flag out of the box because it is specified in [the snapshot file](https://github.com/commercialhaskell/lts-haskell/blob/master/lts-14.4.yaml#L12525-L12526)

**Environment**
 - OS name + version: Linux 4.19.75 #1-NixOS
 - Bazel version: 1.1.0 (from the nix-shell)
 - Version of the rules: 33743d2f (latest master)

/cc @mboes

**Comment by mboes** [comment]
@aherrmann I'm at a loss as to how to resolve this. Stack doesn't expose enough metadata. We'd need 
 to add new subcommands to `stack ls` to expose this information, or augment the new JSON output. Any other ideas?

**Comment by aherrmann** [comment]
Yes, it seems this requires more output from `stack ls dependencies --json`. This makes me wonder, if this becomes available, should `stack_snapshot` still accept Cabal flags, or should users specify those in custom stack snapshot files?

**Comment by mboes** [comment]
Using a custom snapshot file may well be cleaner. Best to keep as much Stack configuration in Stack specific files, rather than have that split between Stack specific files and the `WORKSPACE` file.

**Comment by aherrmann** [comment]
Upstream feature request about exposing the needed metadata: https://github.com/commercialhaskell/stack/issues/5372
